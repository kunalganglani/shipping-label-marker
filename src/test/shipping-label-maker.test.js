import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ShippingLabelMaker from '../features/shipping-label-maker/shipping-label-maker';
import ShippingLabel from '../features/shipping-label-maker/shipping-label';

Enzyme.configure({ adapter: new Adapter() });

describe('ShippingLabelMaker tests', () => {
  let wrapper;
  const props = {
    wizardContext: {
      from: {
        name: 'name',
        street: 'street',
        city: 'city',
        state: 'state',
        zip: 'zip',
      },
      to: {
        name: 'name',
        street: 'street',
        city: 'city',
        state: 'state',
        zip: 'zip',
      },
      weight: 11,
      shippingOption: '1',
    },
  };

  test(' renders correctly', () => {
    const tree = renderer
      .create(<ShippingLabelMaker />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should print a shipping label when createLabel is called', () => {
    wrapper = shallow(<ShippingLabelMaker />);
    wrapper.instance().createLabel(props.wizardContext);
    expect(wrapper.find(ShippingLabel).length).toEqual(1);
  });
  test('handleSender is called', () => {
    const setStateSpy = jest.spyOn(ShippingLabelMaker.prototype, 'setState');
    wrapper = shallow(<ShippingLabelMaker />);
    const mockEvent = {
      target: {
        value: 1,
      },
    };
    wrapper.instance().handleSender(mockEvent);
    expect(setStateSpy).toHaveBeenCalled();
  });

  test('handleReceiver is called', () => {
    const setStateSpy = jest.spyOn(ShippingLabelMaker.prototype, 'setState');
    wrapper = shallow(<ShippingLabelMaker />);
    const mockEvent = {
      target: {
        value: 1,
      },
    };
    wrapper.instance().handleReceiver(mockEvent);
    expect(setStateSpy).toHaveBeenCalled();
  });

  test('handleWeight is called', () => {
    const setStateSpy = jest.spyOn(ShippingLabelMaker.prototype, 'setState');
    wrapper = shallow(<ShippingLabelMaker />);
    const mockEvent = {
      target: {
        value: 1,
      },
    };
    wrapper.instance().handleWeight(mockEvent);
    expect(setStateSpy).toHaveBeenCalled();
  });
  test('handleShipping is called', () => {
    const setStateSpy = jest.spyOn(ShippingLabelMaker.prototype, 'setState');
    wrapper = shallow(<ShippingLabelMaker />);
    const mockEvent = {
      target: {
        value: 1,
      },
    };
    wrapper.instance().handleShipping(mockEvent);
    expect(setStateSpy).toHaveBeenCalled();
  });

  test('isReceiverValid is called', () => {
    wrapper = shallow(<ShippingLabelMaker />);
    expect(wrapper.instance().isReceiverValid()).toEqual(false);
  });
});
