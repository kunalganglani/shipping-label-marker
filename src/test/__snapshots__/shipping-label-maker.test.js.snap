// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingLabelMaker tests  renders correctly 1`] = `
<div>
  <div
    className="css-1frs77m"
  >
    <h2>
      Shipping Label Wizard
    </h2>
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={20}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-80%)",
          }
        }
      />
    </div>
    <div
      className="css-3q9rh4"
    >
      <h3>
        Enter the sender's address
      </h3>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="name"
          id="name-label"
        >
          Name
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="street"
          id="street-label"
        >
          Street
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="street"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Street
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="css-14dkja1"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="city"
            id="city-label"
          >
            City
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="city"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  City
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="state"
            id="state-label"
          >
            State
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="state"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  State
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="zip"
            id="zip-label"
          >
            Zip
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="zip"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="number"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Zip
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1w23u0l"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Next
        </span>
      </button>
    </div>
  </div>
</div>
`;
